src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/WritableBufferWriter.cs

```diff
@@ -0,0 +1,360 @@
+﻿using System;
+using System.IO.Pipelines;
+using System.Runtime.CompilerServices;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Internal.Infrastructure
+{
+    public struct WritableBufferWriter
```

Comment:
`BufferWriter` unless you are keeping the door open for writing to a NonWritableBuffer 😜